src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,162 @@
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println("[" + list.get(index).getStatusIcon() + "] " + list.get(index).description);
+            break;
+        case "todo":
+            if (description==null) {
+                throw new DukeException("nullTodo");
+            }
+            count++;
+            Todo todo = new Todo(description, count);
+            printAdded(todo,count);
+            list.add(todo);
+            break;
+        case "deadline":
+            if (description==null) {
+                throw new DukeException("nullTodo");
+            } else if (time==null) {
+                throw new DukeException("nullTime");
+            }
+            count++;
+            Deadline deadline = new Deadline(description, count, time);
+            printAdded(deadline,count);
+            list.add(deadline);
+            break;
+        case "event":
+            if (description==null) {
+                throw new DukeException("nullTodo");
+            } else if (time==null) {
+                throw new DukeException("nullTime");
+            }
+            count++;
+            Event event = new Event(description, count, time);
+            printAdded(event,count);
+            list.add(event);
+            break;
+        case "bye":
+            System.out.println("Bye. Hope to see you again soon!");
+            return -1;
+        case "delete":
+            if (description==null) {
+                throw new NumberFormatException();
+            }
+            int index2 = Integer.parseInt(description)-1;
+            list.get(index2).printTaskDelete(count-1);
+            list.remove(index2);
+            count--;
+            for (int i=index2; i<count; i++) {
+                list.get(i).index--;
+            }
+            break;
+        default:
+            throw new DukeException("unknownCommand");
+        }
+        return 0;
+    }
+
+    public void run() {
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        in = new Scanner(System.in);
+        String input = in.nextLine();
+        while (true) {
+            try {
+                if (executeCommand(input) == -1) {
+                    break;
+                }
+            } catch (DukeException e) {
+                switch (e.error) {
+                case "unknownCommand":
+                    System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                    break;
+                case "nullTodo":
+                    System.out.println("☹ OOPS!!! The description cannot be empty.");
+                    break;
+                case "nullTime":
+                    System.out.println("☹ OOPS!!! The time cannot be empty.");
+                    break;
+                default:
+                    System.out.println("☹ OOPS!!! There is an unknown error.");
+                    break;
+                }
```

Comment:
better to have multiple types of exceptions and put each in its own catch block

Reply:
I used the main exception DukeException as I needed to throw multiple exceptions from a method